Ghidra/Framework/Generic/src/main/java/ghidra/util/NumericUtilities.java

```diff
@@ -92,6 +92,52 @@ public static Long parseNumber(String numStr, Long defaultValue) {
 		return value;
 	}
 
+	/**
+	 * parses the given string as a numeric value, detecting whether or not it begins with a Hex
+	 * prefix, and if not, parses as a int value.
+	 */
+	public static int parseInt(String numStr) {
+		String origStr = numStr;
+		int value = 0;
+		int sign = 1;
+
+		numStr = (numStr == null ? "" : numStr.trim());
+		if (numStr.length() == 0) {
+			return value;
+		}
+		if (numStr.startsWith("-")) {
+			sign = -1;
+			numStr = numStr.substring(1);
+		}
+		int radix = 10;
+
+		if (numStr.startsWith(HEX_PREFIX_x) || numStr.startsWith(HEX_PREFIX_X)) {
+			if (numStr.length() > 10) {
+				throw new NumberFormatException(numStr + " has too many digits.");
+			}
+			numStr = numStr.substring(2);
+			radix = 16;
+		}
+		if (numStr.length() == 0) {
+			return 0;
+		}
```

Comment:
This seems wrong to me. This if-case leads to `parseInt("0x")`, `parseInt("-")` and `parseInt("-0x")` to all return `0`, instead of throwing a `NumberFormatException`. Throwing a `NumberFormatException` seems more appropriate here, since the input is not a correctly formatted integer.